src/_pytest/fixtures.py

```diff
@@ -1303,6 +1293,15 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
                     name = marker.name
                 assert not name.startswith(self._argprefix), FIXTURE_MSG.format(name)
 
+            # during fixture definition we wrap the original fixture function
+            # to issue a warning if called directly, so here we unwrap it in order to not emit the warning
+            # when pytest itself calls the fixture function
+            if six.PY2 and unittest:
```

Comment:
This was the best I could come with; unbound methods are a headache to deal here.

Reply:
cant wait for the python2 drop ^^

Reply:
You and me both! :grin: